### ISSUE TITLE
there is no main function in preprocessing_masks.py file

### ISSUE THREAD

**Issue by young-pataro** [naimsassine/MasterThesisInfo2021#2]
Hey Naim,
I am replicating your work and have found that there is no main function in preprocessing_masks.py file，I would greatly appreciate it if you could help me.

**Comment by naimsassine** [comment]
Hello! There is no main function because it steps of the pipeline to obtain the training data can be different depending on how you use the data! But all the functions that are in the preprocessing file can help you do that. If you want to reproduce exactly what I did, you can follow the steps in the pdf file that you can find in the repo (normally these steps are functions that are already implemented :))
Hope this helps!

**Comment by young-pataro** [comment]
&nbsp; &nbsp; &nbsp;Thank you very much for replying to me in your spare time. I have read your paper many times and would like to ask you a few questions.
&nbsp; &nbsp; &nbsp;Firstly, the dataset for prostate segmentation is Promise 12, while the dataset for prostate cancer segmentation is Prostate X. Why not use the Prostate X dataset for both segmentation stages?
&nbsp; &nbsp; Secondly, due to the different datasets of the two segmentation tasks, how is the connection process achieved by using the mask size of the first stage of prostate segmentation to crop the prostate cancer image and then using the cropped image as input for the second stage.

&nbsp; &nbsp;&nbsp;I hope to receive your answer again！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "naimsassine/Application-of-Deep-Learning-to-Prostate-Cancer"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月13日(星期六) 下午5:14
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [naimsassine/Application-of-Deep-Learning-to-Prostate-Cancer] there is no main function in preprocessing_masks.py file (Issue #2)





 
Hello! There is no main function because it steps of the pipeline to obtain the training data can be different depending on how you use the data! But all the functions that are in the preprocessing file can help you do that. If you want to reproduce exactly what I did, you can follow the steps in the pdf file that you can find in the repo (normally these steps are functions that are already implemented :))
 Hope this helps!

**Comment by naimsassine** [comment]
Hello!
So for the first questions, the Prostate 12 dataest doesnt' contain cancer segmentation masks and the Prostate X dataset doesn't contain prostate segmentation masks, unless these datasets changed from two years ago (when I wrote the paper). But indeed if you find a dataset with both masks, it wouldn't make sense to use two different ones like I did :)
For the second question, I'll try to explain the flow on a high level : 
- I built one model to segment the prostate in an MRI scan using the Prostate X
- To build the second model, I needed the training set of the promise 12 dataset. But I used the first model I build to improve the promise 12 dataset by zooming in on the prostate and clearing some noise around the prostate.
- Now, how did I use the first model to zoom in on the prostate : I segment the prostate of the promise 12 using the first model, then I take the center of the mask and I build a fixed size window around it. This allows the training set to be made of same sized images of the prostate !
I hope its clear :)

**Comment by young-pataro** [comment]
Thank you very much for your clear answer.


Yes, currently there are publicly available masks for prostate segmentation and prostate cancer segmentation in the Prostate X dataset, and the file format is nii.gz. Therefore, I would like to use T2 images from the Prostate X dataset to complete the segmentation of these two stages. Do you have any recommended code for cascading segmentation? Not finding the relevant learning code made me very confused.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "naimsassine/Application-of-Deep-Learning-to-Prostate-Cancer"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月14日(星期天) 晚上9:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [naimsassine/Application-of-Deep-Learning-to-Prostate-Cancer] there is no main function in preprocessing_masks.py file (Issue #2)





 
Hello!
 So for the first questions, the Prostate 12 dataest doesnt' contain cancer segmentation masks and the Prostate X dataset doesn't contain prostate segmentation masks, unless these datasets changed from two years ago (when I wrote the paper). But indeed if you find a dataset with both masks, it wouldn't make sense to use two different ones like I did :)
 For the second question, I'll try to explain the flow on a high level :
  
I built one model to segment the prostate in an MRI scan using the Prostate X
 
To build the second model, I needed the training set of the promise 12 dataset. But I used the first model I build to improve the promise 12 dataset by zooming in on the prostate and clearing some noise around the prostate.
 
Now, how did I use the first model to zoom in on the prostate : I segment the prostate of the promise 12 using the first model, then I take the center of the mask and I build a fixed size window around it. This allows the training set to be made of same sized images of the prostate !
 I hope its clear :)

**Comment by naimsassine** [comment]
Oh alright nice, this must have changed since 2021 then :)
Mmmmm to be honest I think many techniques can work, but since we are talking about the same types of images and kind of the same segmentation, maybe try transfer learning? 
If I were you (and if you had the time of course), I would first go with two separate models to see accurate they could be, and then move on to a more complex pipeline, like transfer learning or something else!
